api/data/refactoring.py

```diff
@@ -242,3 +245,134 @@ def save_lovd_as_vcf(data, save_to="./lovd.vcf"):
+            populations
+            {{
+              id
+              ac
+              an
+            }}
+          }}
+          genome
+          {{
+            ac
+            an
+            ac_hom
+            populations
+            {{
+              id
+              ac
+              an
+            }}
+          }}
+        }}
+      }}
+    }}
+    """
+
+    response = requests.post(url, json={'query': query}, timeout=300)# timeout set to 5 minutes
+
+    if response.status_code != 200:
+        print('Error:', response.status_code)
+        return None
+
+    data = response.json()['data']['gene']['variants']
+
+    df = pd.json_normalize(data)
+
+    df['total_ac'] = df['exome.ac'].fillna(0) + df['genome.ac'].fillna(0)
+    df['total_an'] = df['exome.an'].fillna(0) + df['genome.an'].fillna(0)
+
+    df['HGVS Consequence'] = df['hgvsc'].fillna(0) # cDNA change
+    df['Protein Consequence'] = df['hgvsp'].fillna(0) # Protein change
+
+    df['Allele Frequency'] = df['total_ac'] / df['total_an']
+    df['Homozygote Count'] = df['exome.ac_hom'].fillna(0) + df['genome.ac_hom'].fillna(0)
+    exome_populations = df['exome.populations']
+    genome_populations = df['genome.populations']
+    ids = ['afr', 'eas', 'asj', 'sas', 'nfe', 'fin', 'mid', 'amr', 'ami', 'remaining']
+
+    for i in range(len(exome_populations)):
+        exome_pop = exome_populations[i]
+        process_population_data(df, exome_pop, 'exome', ids, i)
+        genome_pop = genome_populations[i]
+        process_population_data(df, genome_pop, 'genome', ids, i)
+
+    for variant_id in ids:
+        df[f'Allele_Frequency_{variant_id}'] = (df[f'exome_ac_{variant_id}'].fillna(0) + df[f'genome_ac_{variant_id}'].fillna(0)) / (
+                        df[f'exome_an_{variant_id}'].fillna(0) + df[f'genome_an_{variant_id}'].fillna(0))
+    population_mapping = {
+            'afr': 'African/African American',
+            'eas': 'East Asian',
+            'asj': 'Ashkenazi Jew',
+            'sas': 'South Asian',
+            'nfe': 'European (non-Finnish)',
+            'fin': 'European (Finnish)',
+            'mid': 'Middle Eastern',
+            'amr': 'Admixed American',
+            'ami': "Amish",
+            'remaining': 'Remaining',
+            '': ''
+        }
+    for i in range(len(df)):
+        max_pop = 0
+        maxid = ''
+        for variant_id in ids:
+            if df.loc[i, f'Allele_Frequency_{variant_id}'] > max_pop:
+                max_pop = df.loc[i, f'Allele_Frequency_{variant_id}']
+                maxid = variant_id
+        df.loc[i, 'Popmax'] = max_pop
+        df.loc[i, 'Popmax population'] = population_mapping[maxid]
+    not_to_drop = ['Popmax', 'Popmax population', 'Homozygote Count', 'Allele Frequency', 'variant_id',
+                       'cDNA change', 'Protein change']
+    df = df.drop([col for col in df.columns if col not in not_to_drop], axis=1)
+
+    df.rename(columns={'variant_id': 'gnomAD ID'}, inplace=True)
```

Comment:
In first line you do not use inplace, for second you use. What is the difference?

Reply:
inplace = true keeps the file, inplace = false copies file and assings it. I removed it initially since it doesnt show any benefit